### PR TITLE
Add dist-tag functionality

### PR DESCRIPTION
**Pull request by paulsevere**
Adds a new command line flag and publish config option to enable npm publish 'dist-tag' functionality.

- `tag` (string): publish the package with a dist-tag  (see [npm-dist-tag](https://docs.npmjs.com/cli/dist-tag) ).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
 # 1.0.0
 
 - Initial version of `safe-publish` script.
+
+# 1.1.0
```
Comment (RecuencoJones) [inline CHANGELOG.md]: We usually put the latest version at the top of the file, could you please move this to the top?

File: publish.js [hunk 2]
```diff
@@ -44,44 +44,55 @@ function find({ cwd, name, version, registry }) {
...
+        cwd,
+        registry,
+        tag,
+        dryRun
+      }))
     .then((code) => {
       if (code === resultCode.NEW_VERSION) {
-        log.info(`Package ${ bold(blue(name)) } has been updated to v${ bold(blue(version)) }`);
+        log.info(
+          `Package ${ bold(blue(name)) } has been updated to v${ bold(blue(version)) } ${ (tag &&
```
Comment (RecuencoJones) [inline publish.js]: Instead of `a && b || c` let's use `a ? b : c` as you did on `bin/cli.js#33`

**Review by RecuencoJones — CHANGES_REQUESTED** [review]
Thanks @paulsevere for your contribution!

I added a couple of comments, but overall it LGTM!

File: publish.js (hunk 2 above)
Comment (paulsevere) [reply publish.js]: Sure will do.

File: CHANGELOG.md (hunk 1 above)
Comment (paulsevere) [reply CHANGELOG.md]: Sure will do.

**Comment by RecuencoJones** [comment]
Build failed due to some linter issues:

```
/home/travis/build/adidas/safe-publish/publish.js
  11:17  error  Unexpected trailing comma                            comma-dangle
  23:37  error  Expected space(s) after '${'                         template-curly-spacing
  23:47  error  Expected space(s) before '}'                         template-curly-spacing
  25:25  error  Expected space(s) after '${'                         template-curly-spacing
  25:31  error  Expected space(s) before '}'                         template-curly-spacing
  25:33  error  Expected space(s) after '${'                         template-curly-spacing
  25:42  error  Expected space(s) before '}'                         template-curly-spacing
  25:52  error  Expected space(s) after '${'                         template-curly-spacing
  25:68  error  Expected space(s) before '}'                         template-curly-spacing
  31:12  error  Expected parentheses around arrow function argument  arrow-parens
  50:27  error  Expected space(s) after '${'                         template-curly-spacing
  50:32  error  Expected space(s) before '}'                         template-curly-spacing
  51:37  error  Expected space(s) after '${'                         template-curly-spacing
  51:47  error  Expected space(s) before '}'                         template-curly-spacing
  54:28  error  Expected space(s) after '${'                         template-curly-spacing
  54:44  error  Expected space(s) before '}'                         template-curly-spacing
  61:11  error  Expected parentheses around arrow function argument  arrow-parens
  63:27  error  Expected space(s) after '${'                         template-curly-spacing
  63:39  error  Expected space(s) before '}'                         template-curly-spacing
  69:11  error  Expected parentheses around arrow function argument  arrow-parens
  75:15  error  Unexpected trailing comma                            comma-dangle
  77:5   error  Unexpected newline before '('                        function-paren-newline
  78:11  error  Expected parentheses around arrow function argument  arrow-parens
  81:20  error  Expected space(s) after '${'                         template-curly-spacing
  81:38  error  Expected space(s) before '}'                         template-curly-spacing
  81:61  error  Expected space(s) after '${'                         template-curly-spacing
  81:82  error  Expected space(s) before '}'                         template-curly-spacing
  82:28  error  Expected space(s) after '${'                         template-curly-spacing
  82:46  error  Expected space(s) before '}'                         template-curly-spacing
  87:20  error  Expected space(s) after '${'                         template-curly-spacing
  87:39  error  Expected space(s) before '}'                         template-curly-spacing
  87:64  error  Expected space(s) after '${'                         template-curly-spacing
  87:86  error  Expected space(s) before '}'                         template-curly-spacing
  88:28  error  Expected space(s) after '${'                         template-curly-spacing
  88:46  error  Expected space(s) before '}'                         template-curly-spacing
  93:12  error  Expected parentheses around arrow function argument  arrow-parens
  95:27  error  Expected space(s) after '${'                         template-curly-spacing
  95:44  error  Expected space(s) before '}'                         template-curly-spacing
  95:50  error  Expected space(s) after '${'                         template-curly-spacing
  95:70  error  Expected space(s) before '}'                         template-curly-spacing
  96:26  error  Expected space(s) after '${'                         template-curly-spacing
  96:42  error  Expected space(s) before '}'                         template-curly-spacing
```

They can all be fixed by running eslint with `--fix`

**Comment by paulsevere** [comment]
I fixed those linter errors and addressed your comments (I think) .. all set?

**Review by RecuencoJones — APPROVED** [review]

**Comment by moelders** [comment]
Thanks @paulsevere! We are gonna test it and release the version.

**Comment by moelders** [comment]
Version 1.1.0 just published.

**Pull request merged**